geomstats/visualization/symmetric_positive_definite.py

```diff
@@ -0,0 +1,598 @@
+
+        Returns
+        -------
+        array-like, shape [2,2]
+        """
+        x, y, z = point
+        return np.array([[z + x, y], [y, z - x]])
+
+    @staticmethod
+    def spd_to_xyz(matrix):
+        """Convert coordinates on manifold to Cartesian coordinates.
+
+        Parameters
+        ----------
+        point : tuple-like of size = 3
+
+        Returns
+        -------
+        tuple-like of size = 3
+        """
+        z = (matrix[0, 0] + matrix[1, 1]) / 2.0
+        x = matrix[0, 0] - z
+        y = matrix[0, 1]
+
+        return (x, y, z)
+
+    def find_color_for_point(self, point):
+        """Convert the color from HSV coordinates to RGB coordinates.
+
+        Parameters
+        ----------
+        point : tuple-like of size = 3
+
+        Returns
+        -------
+        color: tuple-like of size = 3
+        """
+        x, y, z = point
+
+        # convert radians to degrees
+        angle = np.arctan2(x, y) * 180 / np.pi
+
+        # normalize degrees to [0, 360]
+        if angle < 0:
+            angle = angle + 360
+
+        hue = angle / 360
+        saturation = math.sqrt(x**2 + y**2) / self.max_z
+        value = z / self.max_z
+
+        color = colorsys.hsv_to_rgb(hue, saturation, value)
+
+        return color
+
+    def plot_grid(self):
+        """Plot the geodesic grid.
+
+        Returns
+        -------
+            Figure plot
+        """
+        self.plot_geodesic(
+            startpt_xyz=(0, 0, 0.5),
+            endpt_xyz=(0, 0, 0.6),
+            n_geodesic_samples=30
+        )
+        self.plot_geodesic(
+            startpt_xyz=(0, 0, 0.5),
+            endpt_xyz=(0, 0.1, 0.5),
+            n_geodesic_samples=30
+        )
+        self.plot_geodesic(
+            startpt_xyz=(0, 0, 0.5),
+            endpt_xyz=(0.1, 0, 0.5),
+            n_geodesic_samples=30
+        )
+
+    def plot_rendering_top(self, n_radii, n_angles):
+        """Plot ellipses on the top of the cone manifold.
+
+        Parameters
+        ----------
+        n_angles : int
+            Number of angles in polar coordinates
+        n_radii : int
+            Number of radii in polar coordinates
+
+        Returns
+        -------
+        Figure plot
+        """
+        # Does not include radius r=0, this is to eliminate duplicate points
+        # z_plane = self.max_z
+
+        z_plane = self.currZ
+        radii = np.linspace(z_plane, 0, n_radii, endpoint=False)
+
+        # An array of angles
+        angles = np.linspace(0, 2 * np.pi, n_angles, endpoint=False)
+
+        # Repeat all angles for each radius
+        angles = np.repeat(angles[..., np.newaxis], n_radii, axis=1)
+
+        x = np.append(0, (radii * np.cos(angles)).flatten())
+
+        y = np.append(0, (radii * np.sin(angles)).flatten())
+
+        for x_tmp, y_tmp in zip(x, y):
+            sampled_xyz = (x_tmp, y_tmp, z_plane)
+            sampled_spd = SymPosDefVisualization.xyz_to_spd(sampled_xyz)
```

Comment:
sampled_spd = self.xyz_to_spd(sampled_xyz)
--> don't use the class, use self instead